packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -2,6 +2,27 @@ import cxapi = require('@aws-cdk/cx-api');
 import { ConstructOrder, Root } from './construct';
 import { FileSystemStore, InMemoryStore, ISynthesisSession, SynthesisSession } from './synthesis';
 
+/**
+ * Custom construction properties for a CDK program
+ */
+export interface AppProps {
+  /**
+   * Automatically call run before the application exits
+   *
+   * If you set this, you don't have to call `run()` anymore.
+   *
+   * @default true if running via CDK toolkit, false otherwise
```

Comment:
Be more specific about the mechanism: `true if running via the toolkit (CDK_OUTDIR is set)`